orc8r/cloud/go/service/middleware/unary/interceptors.go

```diff
@@ -76,29 +76,17 @@ func errlogInterceptor(ctx context.Context, req interface{}, info *grpc.UnarySer
 	return res, err
 }
 
-// setIdentityInterceptor sets the caller's identity from certificate metadata
-// in the caller's context.
-// TODO(hcgatewood): remove this wrapper and just use SetIdentityFromContext directly
-func setIdentityInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
-	newCtx, _, res, err := SetIdentityFromContext(ctx, req, info)
-	if err != nil {
-		return res, err
-	}
-	if newCtx != nil {
-		ctx = newCtx
-	}
-	return handler(ctx, req)
-}
+type decorator func(context.Context, interface{}, *grpc.UnaryServerInfo) (context.Context, interface{}, interface{}, error)
```

Comment:
Isn't this the type of function you wish to decorate rather than the decorator itself? I believe what's currently called `makeInterceptorFromDecorator` is actually the decorator, as it's the thing that wraps around the functions SetIdentityFromContext/BlockUnregisteredGateways.

Reply:
Maybe interceptorTemplate is a better name than decorator (and subsequently `makeInterceptorFromTemplate`)